### DISCUSSION TITLE
ssd1309 banding/ghosting issue

### DISCUSSION THREAD

**Discussion by PsuFan** [micropython/micropython#11863] [Libraries & Drivers]
I was hoping someone could shed some light on this issue, not sure if its a bad display or driver. I am using the micropython ssd1306 for a ssd1309 display... The banding gets worse depending on what is showing and seems particularly bad with text. It also seems somewhat unrepeatable if you look at some of the line tests. See pic, thanks!
https://www.amazon.com/dp/B09LND6QJ1

![Display](https://github.com/micropython/micropython/assets/5677263/7ffc7f65-d111-458b-b6d6-84f54cbf2bcf)

**Comment by bidrohini** [comment]
It can be a bad display. I think you have to check it separately. Do you have an Arduino? If yes, you can easily test your OLED display with that. You have to use SSD1306.h library. If the problem persists, you can be quite sure that the display has some problems in it. You can also check if there is a loose connection somewhere in your circcuit. Here is a good project on Raspberry Pi Pico and SSD1306: 

https://www.pcbway.com/project/shareproject/Raspberry_Pico_with_Ultrasonic_Sensor_and_SSD1306.html 

The code is written in micropython. You can try this to cross-check your display.

**Comment by PsuFan** [comment]
I am using the micropython ssd1306 driver already. I just swapped the display by using a brand new one in the ribbon cable. Still using the same display board because I had to jump 4 pads to get on I2C. No change.

**Comment by PsuFan** [comment]
There is no change 3v3 vs 5v. There is a significant change setting the external_vcc True. Bands are much worse enabling this setting.

**Comment by massimosala** [comment]
Last test... is it wired thru i2c or spi ? Try to change the interface and to lower the bus speed.

But unfortunately it seems a broken display.

**Reply by PsuFan** [reply]
I2C. I plan to test spi. I’ve swapped the actual display itself and it was exactly the same with a different screen. Used the same display board. I have one board setup for I2C and an unmodified one for spi that I will have to check. Seems like it could be a driver issue because the external_vcc setting drastically made it worse, will look into ssd1309 settings like contrast etc…

**Reply by PsuFan** [reply]
Not sure where to go now, tested the spi display with https://github.com/rdagger/micropython-ssd1309 and it works fine, no artifacts. Not sure if its a driver or display board issue switching to i2c. I'll look at the driver registers even more closely to see if there are any differences.

**Comment by PsuFan** [comment]
Ugh I changed VCOM_DESELECT_LEVEL from 0x30 to 0x40 and removed IREF_SELECT 0x30 and now it appears to be working great so far...

Edit: no idea whats going on, was working now its back to not working...
Edit2: hard reset fixed it, anyone else that runs into this set vcom 0x40 and remove iref_select. Leaning more towards the iref_select.
